src/test/groovy/edu/ie3/simona/model/participant/ChpModelTest.groovy

```diff
@@ -163,23 +169,21 @@ class ChpModelTest extends Specification {
 
     when:
     chpModel.calculateNextState(chpData)
-    def storageLevel = CylindricalThermalStorage.energyToVolume(thermalStorage._storedEnergy(), thermalStorage.c(), thermalStorage.inletTemp(), thermalStorage.returnTemp())
-    def resStorageLvl = storageLevel.to(CUBIC_METRE)
 
     then:
-    equals(storageLevel, resStorageLvl, TOLERANCE)
+    Math.abs(thermalStorage._storedEnergy().toKilowattHours() - expectedStoredEnergy.doubleValue()) < TOLERANCE
 
     where:
-    chpState           | storageLvl | heatDemand | expectedStorageLevel
-    chpStateNotRunning | 90         | 0         || 90                    // tests case (false, false, true)
-    chpStateNotRunning | 90         | 8 * 115   || 20                // tests case (false, true, false)
-    chpStateNotRunning | 90         | 10        || 89.1304                // tests case (false, true, true)
-    chpStateRunning    | 90         | 0         || 98.6956                    // tests case (true, false, true)
-    chpStateRunning    | 90         | 8 * 115   || 20                // tests case (true, true, false)
-    chpStateRunning    | 90         | 10        || 97.8260                // tests case (true, true, true)
-    chpStateRunning    | 90         | 806       || 28.6086                // test case (_, true, false) and demand covered together with chp
-    chpStateRunning    | 90         | 9 * 115   || 20                // test case (_, true, false) and demand not covered together with chp
-    chpStateRunning    | 92         | 1         || 100                        // test case (true, true, true) and storage volume exceeds maximum
+    chpState           | storageLvl | heatDemand | expectedStoredEnergy
+    chpStateNotRunning | 90         | 0         || 1035                    // tests case (false, false, true)
+    chpStateNotRunning | 90         | 8 * 115   || 230                // tests case (false, true, false)
+    chpStateNotRunning | 90         | 10        || 1025                // tests case (false, true, true)
+    chpStateRunning    | 90         | 0         || 1135                    // tests case (true, false, true)
+    chpStateRunning    | 90         | 8 * 115   || 230                // tests case (true, true, false)
+    chpStateRunning    | 90         | 10        || 1125               // tests case (true, true, true)
+    chpStateRunning    | 90         | 806       || 329                // test case (_, true, false) and demand covered together with chp
+    chpStateRunning    | 90         | 9 * 115   || 230                // test case (_, true, false) and demand not covered together with chp
+    chpStateRunning    | 92         | 1         || 1150                        // test case (true, true, true) and storage volume exceeds maximum
```

Comment:
Values changed because now the stored energy is checked instead of storageLevel. This was a workaround to avoid the local creation of additional methods that arent covert in the squants package. But since you and Marius brought up valuable reasons to include them as well this could be rolled back to original test version.